### DISCUSSION TITLE
Can't build Wasm project with SKXmlCanvas control - Visual Studio 2022 Preview

### DISCUSSION THREAD

**Discussion by ricardoromaobr** [unoplatform/uno#6918] [Q&A]
# Trying to build uno app with skiasharp view
I'm trying to experiment a draw with unoplatform. 
I create a project to make some experiments;
I'm running Uwp project and clicking on canvas and drawing a red circle on the surface. 
But when I try do compile the Wasm project an error to build the project is showed:
![image](https://user-images.githubusercontent.com/7851330/131439236-e7561187-4f9e-442e-83ed-a9ff635bea1e.png)

I started with Uno.Wasm.Bootstrap.DevServer and Uno.Wasm.Bootstrap from version 2.10 
Some version has a problem to found the linux subsystem. From version 3.0.0-dev-101 this error start showed. 

The repository with the small test app is [here](https://github.com/ricardoromaobr/MyApp4.git)

The uno-check below: 
![image](https://user-images.githubusercontent.com/7851330/131440145-8fbc4d43-ba02-4b82-9711-e72491874084.png)

I create the app using dotnet CLI
dotnet new unoapp -o MyApp4

**Comment by jeromelaban** [comment]
Thanks for the report. You may need to take a look at the binlog to troubleshoot the error: https://platform.uno/docs/articles/uno-builds-troubleshooting.html

**Reply by ricardoromaobr** [reply]
What do I do with the msbuild.binlog?

**Reply by ricardoromaobr** [reply]
The msbuild.binlog
[msbuild.binlog.zip](https://github.com/unoplatform/uno/files/7083782/msbuild.binlog.zip)

**Reply by jeromelaban** [reply]
Thanks. It seems that the python version installed on your system is too old (or missing modules) for emscripten to work with. Can you instal a more recent python version on your system [Python from Windows Store](https://www.microsoft.com/store/productId/9P7QFQMJRFP7), or manually through [Python's official site](https://www.python.org/downloads/).

**Reply by ricardoromaobr** [reply]
The same error continue
The Python version
![image](https://user-images.githubusercontent.com/7851330/131595547-4414bfc5-44c2-4490-b0be-84cc4b735e6a.png)

And the compile error
![image](https://user-images.githubusercontent.com/7851330/131595724-97169c16-e068-4ed9-aadb-44aaa2a9c6e2.png)

I will try to do it on my  macbook pro with vs code if works I will use them.

**Reply by jeromelaban** [reply]
You're likely getting a different error this time. Take a look at a new binlog to see what is not working.

**Reply by ricardoromaobr** [reply]
I tried more one time.
[msbuild.zip](https://github.com/unoplatform/uno/files/7098629/msbuild.zip)

**Reply by ricardoromaobr** [reply]
I tried to use a stable dotnet 5.0.400 too. 
I used the Uno.Wasm.Bootstrap.DevServer  and  Uno.Wasm.Bootstrap 2.1.0
The project compile and run. But the events not works.

**Comment by jeromelaban** [comment]
From your build log, it seems that emscripten has failed to initialize. Can you try deleting the folder `C:/Users/ricardo/.uno`  and try again with the `3.0.0-dev.138` bootstrapper?

Also, for the pointers to work, currently in Uno the `Canvas` needs to have non-null background, like this:

```xaml
<skia:SKXamlCanvas   
                   PointerMoved="Rectangle_PoiterMoved" 
                   PointerPressed="Rectangle_PoiterPressed"
                   PaintSurface="OnPaintSurface"
                   x:Name="Workbench" 
                   Background="Transparent"
                   Grid.Row="1"/>

```

**Reply by ricardoromaobr** [reply]
Ok. 
Visual Studio 2019 withUno.Wasm.Bootstrap.DevServer version 2.1.0 works fine.
Visual studio 2022 found wsl only with the version of Uno.Wasm.Bootstrap.DevServer is 3.0.0-dev.138.

Build failed using  viusal studio 20222. 
But if I use de CLI donet build it works. 
[Visual Studio 2022_output_builder.txt](https://github.com/unoplatform/uno/files/7105725/Visual.Studio.2022_output_builder.txt)
I would build using CLI and debugger with visual studio 2022 I will try in that way for now. 
I preferer using visual studio 2022 preview because it show less fake errors that visual studio 2019.
Looking in the output I see that visual studio 2022 use a python in the .uno directory and  not  the installed  by me. Is correct  this observation?
With that I can make a better experiment
Thanks very much!

**Reply by jeromelaban** [reply]
It seems that emscripten is not working well on your windows installation, for some reason. Could be related to the existing python installation you had.

You can still for the use of WSL by adding this to your project:

```xml
<PropertyGroup>
   <WasmShellEnableEmscriptenWindows>true</WasmShellEnableEmscriptenWindows>
</PropertyGroup>

**Comment by ricardoromaobr** [comment]
The result from CLI
![image](https://user-images.githubusercontent.com/7851330/132003918-3c8c9089-ba0e-485c-a7fa-c43b2180d224.png)


My dotnet frameworks installed
![image](https://user-images.githubusercontent.com/7851330/132004183-4fc4f723-6d6a-47fb-8b36-22211d1a0bc9.png)

**Reply by jeromelaban** [reply]
Please answer in the original thread, each new section is flagged as a possible answer :)

**Reply by ricardoromaobr** [reply]
> It seems that emscripten is not working well on your windows installation, for some reason. Could be related to the existing python installation you had.
> 
> You can still for the use of WSL by adding this to your project:
> 
> ```
> <PropertyGroup>
>    <WasmShellEnableEmscriptenWindows>true</WasmShellEnableEmscriptenWindows>
> </PropertyGroup>
> ```

I set the PHYTHONHOME environment variable  and add this new PropertyGroup on project. With that the visual studio 2022 works. 
Thanks! Now I will take some time to finish my tests.
